lib/bolt/inventory/inventory.rb

```diff
@@ -209,16 +210,46 @@ def create_target_from_hash(data)
         end
 
         if (aliases = new_target.target_alias)
+          @groups.insert_alia(new_target.name, aliases)
+        end
+
+        new_target
+      end
+
+      def validate_target_from_hash(target, _existing)
+        groups = Set.new(group_names)
+        targets = target_names
+
+        # Make sure there are no group name conflicts
+        if groups.include?(target.name)
+          raise ValidationError.new("Target name #{target.name} conflicts with group of the same name", nil)
+        end
+
+        # Validate any aliases
+        if (aliases = target.target_alias)
           aliases = [aliases] if aliases.is_a?(String)
           unless aliases.is_a?(Array)
             msg = "Alias entry on #{t_name} must be a String or Array, not #{aliases.class}"
             raise ValidationError.new(msg, @name)
           end
-
-          @groups.insert_alia(new_target.name, aliases)
         end
 
-        new_target
+        # Make sure there are no conflicts with the new target aliases
+        aliases = @groups.target_aliases
+        target.target_alias.each do |alia|
+          if groups.include?(alia)
+            raise ValidationError.new("Alias #{alia} conflicts with group of the same name", nil)
+          end
+
+          if targets.include?(alia)
+            raise ValidationError.new("Alias #{alia} conflicts with target of the same name", nil)
+          end
+
+          next unless aliases[alia] && aliases[alia] != target.name
```

Comment:
This should just be another `if aliases[alia] && aliases[alia] != target.name`

Reply:
Rubocop is forcing this with our current cops. I'll get that updated.